### PR TITLE
Implement test for Android

### PR DESCRIPTION
**Pull request by rockapot**
Support running the `BuildProject` test on Android, and add a Github Action for it.

Most of the contents under BuildProject/Android was autogenerated using `react-native init BuildProject`.

Also rename existing Github Actions, and make the npm deploy action happen on tag instead of push. The npm deploy action doesn't work anyways, so we'll either fix it or delete it soon, but we don't need it with every push.

### CODE REVIEW

**Comment by andrewdyates** [comment]
@dtdennis28 Drew would be best to review this.

**Review by andrewdyates — APPROVED** [review]
approved to unblock if necessary

File: BuildProject/android/gradlew [hunk 1]
```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env sh
+
+#
+# Copyright 2015 the original author or authors.
```
Comment (danbosnichill) [inline BuildProject/android/gradlew:4]: Haha.  Was this auto-generated or copied from somewhere?  Worth checking the license again to make sure it's okay.

File: BuildProject/tests/App.test.js [hunk 2]
```diff
@@ -1,49 +1,57 @@
...
-const config = {
-  platformName: "iOS",
-  platformVersion: "14.4",
-  deviceName: "iPhone 11",
-  app: "build/BuildProject.ipa",
-  automationName: "XCUITest",
-};
-
-const driver = wd.promiseChainRemote('localhost', PORT);
+import wd from 'wd'
```
Comment (danbosnichill) [inline BuildProject/tests/App.test.js:1]: Can you make this test class a .ts too?

**Review by danbosnichill — APPROVED** [review]

File: BuildProject/android/gradlew (hunk 1 above)
Comment (rockapot) [reply BuildProject/android/gradlew:4]: It's Apache 2.0, it's good.

https://www.apache.org/licenses/LICENSE-2.0

Comment (rockapot) [reply BuildProject/android/gradlew:4]: The entire BuildProject/android directory is autogenerated.

File: BuildProject/tests/App.test.js (hunk 2 above)
Comment (rockapot) [reply BuildProject/tests/App.test.js]: Done.

**Comment by rockapot** [comment]
> @dtdennis28 Drew would be best to review this.

I'll get his review later this week, merging now to work on further client integration.

**Pull request merged**